src/scripts/connect-wallet.ts

```diff
@@ -0,0 +1,139 @@
+  else if (client !== null) innerHtml = `<span>${text}</span>`;
+
+  connectButton.innerHTML = innerHtml;
+}
+
+export async function connectWallet(providerKey?: keyof WalletProvider) {
+  const ethereum = (window as ModifiedWindow).ethereum;
+  if (typeof ethereum !== "undefined" && ethereum !== null) {
+    let provider = ethereum;
+
+    if (ethereum.providers?.length && !!providerKey) {
+      provider = ethereum.providers.find((p: WalletProvider) => p[providerKey]);
+    }
+
+    try {
+      updateConnectButtonText("", true);
+      const [account] = await provider.request({ method: "eth_requestAccounts" });
+      client = createWalletClient({
+        account,
+        chain: mainnet,
+        transport: custom(provider),
+      });
+
+      updateConnectButtonText(truncateString(account));
+      wireEvents(provider);
+
+      watchForPrices();
+
+      connectPrompt.classList.remove("visible");
+      whiteContainer.classList.add("visible");
+
+      if (providersModal.open) {
+        providersModal.close();
+      }
+    } catch (error) {
+      updateConnectButtonText("");
+    }
+  }
+}
+
+export async function disconnectWallet() {
+  const ethereum = (window as ModifiedWindow).ethereum;
+  if (typeof ethereum !== "undefined" && ethereum !== null) {
+    let provider = ethereum;
+
+    if (ethereum.providers?.length) {
+      provider = ethereum.providers.find((p: WalletProvider) => p.isCoinbaseWallet || p.isMetaMask);
+    }
+
+    try {
+      await provider.request({ method: "wallet_revokePermissions", params: [{ eth_accounts: {} }] });
+      client = null;
+
+      unwatchForPrices();
+
+      updateConnectButtonText("Connect Wallet");
+      connectPrompt.classList.add("visible");
+      whiteContainer.classList.remove("visible");
+    } catch (error) {
+      updateConnectButtonText("");
+    }
+  }
+}
+
+export async function connectIfAuthorized() {
+  const providerKeys = ["isMetaMask", "isCoinbaseWallet"];
+  const ethereum = (window as ModifiedWindow).ethereum;
+  let provider = ethereum;
+
+  if (typeof ethereum !== "undefined" && ethereum !== null && ethereum.providers?.length) {
+    for (const providerKey of providerKeys) {
+      provider = ethereum.providers.find((p: WalletProvider) => p[providerKey as keyof WalletProvider]);
+
+      if (provider) break;
+    }
+  }
+
+  if (provider) {
+    const accounts: `0x${string}`[] = await provider.request({ method: "eth_accounts" });
+
+    if (accounts.length) {
+      const [account] = accounts;
+      client = createWalletClient({
+        account,
+        chain: mainnet,
+        transport: custom(provider),
+      });
+
+      watchForPrices();
+      updateConnectButtonText(truncateString(account));
+      connectPrompt.classList.remove("visible");
+      whiteContainer.classList.add("visible");
+    }
+  }
+}
+
+//eslint-disable-next-line @typescript-eslint/no-explicit-any
+function wireEvents(provider: any) {
```

Comment:
Here as well. We won't accept with you canceling or changing our rules. They are there for a reason.

Reply:
My apologies. I'll fix it